.github/scripts/create-file-index.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+
+import os
+import time
+import json
+import sys
+import gzip
+from datetime import datetime
+
+def list_files_with_metadata(directory, output_file):
+    # Record the start time
+    start_time = time.time()
+
+    # Get the current date and time for indexing
+    index_timestamp = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+
+    files_metadata = []
+
+    for root, dirs, files in os.walk(directory):
+        for name in files:
+            filepath = os.path.join(root, name)
+            relative_path = os.path.relpath(filepath, directory)
+
+            try:
+                metadata = {
+                    "path": relative_path,
+                    "size": os.path.getsize(filepath),
```

Comment:
this and next ones -- use

```
In [9]: os.stat('123', follow_symlinks=False)
Out[9]: os.stat_result(st_mode=41471, st_ino=1754, st_dev=44, st_nlink=1, st_uid=47521, st_gid=47522, st_size=29, st_atime=1734027506, st_mtime=1734027506, st_ctime=1734027506)
```
instead so works for symlinks. 

for now that should suffice, I do not think we need to store type of file (symlink or not)
BUT check what happens if a directory is a symlink - you should not follow it!!!